gpu-enabler/src/main/scala/com/ibm/gpuenabler/CUDARDDUtils.scala

```diff
@@ -237,7 +240,85 @@ object CUDARDDImplicits {
     extends Serializable {
 
     def sc = rdd.sparkContext
+    
+    def autoCacheGPU(rdd: RDD[_]): Unit = {
+      val lineage = scala.collection.mutable.Set[RDD[_]]()
+      /**
+        * If calling this method in an action before the runJob,
+        * GPUEnabler cache RDDs on GPU automatically
+        * to remove redundant data transfers in a sequent of these
+        * every time GPUEnabler invoke actions has a sequent of GPU methods,
+        *
+        * gpuApiFlag shows the child type of a RDD.
+        * When the child RDD's API
+        *  case 0 => CPU API (ex, mapRDD)
+        *  case 1 => GPU API (don't divide a stage. ex, mapGPURDD)
+        *  case 2 => GPU API (divide a stage into two. ex,shuffleGPURDD)
+        */
+      var gpuApiFlag = 0
+      if(judgeApiType(rdd) == 1) rdd.autoCacheGpu(); gpuApiFlag = 1
```

Comment:
Is this line correct? The current version always set `1` into `gpuApiFlag`.
If this is correct, we can replace line 258 with `val gpuApiFlag = 1`.